### DISCUSSION TITLE
Abort DriveDistance Movement

### DISCUSSION THREAD

**Discussion by tobias-kausch** [iRobotEducation/create3_docs#495] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

5.4

### Which RMW is your robot running?

FastRTPS

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

Hello!
I am trying to give the robot a DriveDistance action goal to drive a specific distance. However, I want to be able to cancel the robot's movement mid drive. For example, I tell it to drive 4 meters but then I decide I want to stop moving at a certain point, to avoid an obstacle for example. How can I do this? Is there any example code of this?

Here is the code I have:

[controller.txt](https://github.com/iRobotEducation/create3_docs/files/13732863/controller.txt)

Does anyone have any suggestions?

**Comment by alsora** [comment]
ROS 2 actions can be cancelled from the client while they are running.
Please refer to the ROS 2 tutorials for this https://github.com/ros2/examples/blob/rolling/rclpy/actions/minimal_action_client/examples_rclpy_minimal_action_client/client_cancel.py
